src/sender.rs

```diff
@@ -14,35 +14,47 @@ use crate::{
 
 pub struct Sender {
     frame_count: FrameCountGenerator,
-    sender_id: KeyId,
+    key_id: KeyId,
     cipher_suite: CipherSuite,
     secret: Option<Secret>,
     buffer: Vec<u8>,
 }
 
 impl Sender {
-    pub fn new(sender_id: u64) -> Sender {
-        Self::with_cipher_suite(sender_id, CipherSuiteVariant::AesGcm256Sha512)
+    pub fn new<K>(key_id: K) -> Sender
+    where
+        K: Into<KeyId>,
+    {
+        Self::with_cipher_suite(key_id, CipherSuiteVariant::AesGcm256Sha512)
     }
 
-    pub fn with_cipher_suite(sender_id: u64, variant: CipherSuiteVariant) -> Sender {
+    pub fn with_cipher_suite<K>(key_id: K, variant: CipherSuiteVariant) -> Sender
+    where
+        K: Into<KeyId>,
+    {
         let cipher_suite: CipherSuite = variant.into();
+        let key_id = key_id.into();
         log::debug!("Setting up sframe Sender");
         log::trace!(
-            "SenderID {} (ciphersuite {:?})",
-            sender_id,
+            "KeyID {:?} (ciphersuite {:?})",
+            key_id,
             cipher_suite.variant
         );
         Sender {
             frame_count: Default::default(),
-            sender_id: sender_id.into(),
+            key_id,
             cipher_suite,
             secret: None,
             buffer: Default::default(),
         }
     }
 
-    pub fn encrypt(&mut self, unencrypted_payload: &[u8], skip: usize) -> Result<&[u8]> {
+    pub fn encrypt<Payload>(&mut self, unencrypted_payload: &Payload, skip: usize) -> Result<&[u8]>
+    where
+        Payload: AsRef<[u8]> + ?Sized,
```

Comment:
are you explicitly using two different terms here? `Payload` vs `Frame`?

Reply:
actually while we're at it, [the standard in this place calls it "plaintext"](https://sframe-wg.github.io/sframe/draft-ietf-sframe-enc.html#name-encryption), which might be a bit more clear

Reply:
I like it :+1: For decrypting I have choosen ÈncryptedFrame` in contrary to `Sframe`, which sounded weird imho